### PR TITLE
DA: Implement base structure for verifier service

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Added verifier service that expects nomos da protocol implementation as a backend.

Libp2p network adapter is reverted to the version as in previous da service implementation - it should be functional, but the plan for a short term is to use http api for tests and cli (until Da Dissemination protocol is defined).

Service integration test will be added after this PR is reviewd and merged.

### CODE REVIEW

File: nomos-da/kzgrs-backend/src/common.rs [hunk 1]
```diff
@@ -54,6 +62,10 @@ impl Column {
     pub fn len(&self) -> usize {
         self.0.len()
     }
+    #[must_use]
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/common.rs:65]: Why must use? (it is repeated in the other methods as well)

File: nomos-services/data-availability/verifier/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            service_state,
+            verifier: Backend::new(verifier_settings),
+        })
+    }
+
+    async fn run(self) -> Result<(), DynError> {
+        // This service will likely have to be modified later on.
+        // Most probably the verifier itself need to be constructed/update for every message with
+        // an updated list of the available nodes list, as it needs his own index coming from the
+        // position of his bls public key landing in the above-mentioned list.
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/verifier/src/lib.rs:150]: It need to update itself. This list need to be read from new blocks if it change.

File: nomos-services/data-availability/verifier/src/network/adapters/libp2p.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    async fn blob_stream(&self) -> Box<dyn Stream<Item = Self::Blob> + Unpin + Send> {
+        self.stream_for::<Self::Blob>().await
+    }
+
+    async fn send_attestation(&self, attestation: Self::Attestation) -> Result<(), DynError> {
+        self.send(attestation).await
+    }
+}
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/verifier/src/network/adapters/libp2p.rs:107]: It is still not sure how would it go. (Probably the approach of the http service we used to have is closer to what we were thinking).
How it is right now may be a problem, how do a node filter what blob is for himself? If he receives all blobs there is no point in doing all the DA dance 😅 

We do not have a DA dispersal protocol yet. But I would say we should start point2point. 
Maybe we bring back the http endpoints for now? wdyt?

Comment (bacv) [reply nomos-services/data-availability/verifier/src/network/adapters/libp2p.rs:107]: > Libp2p network adapter is reverted to the version as in previous da service implementation - it should be functional, but the plan for a short term is to use http api for tests and cli (until Da Dissemination protocol is defined).

I've udes what we had before here. I'm not planning to use it in tests, but will try when reworking cli app. 
Http api will be used for now.

File: nomos-services/data-availability/verifier/src/lib.rs (hunk 2 above)
Comment (bacv) [reply nomos-services/data-availability/verifier/src/lib.rs:150]: (This is your comment) I've also added another one where the verifier is initialized. I'll work on "params provider" service after basic integration between services suceeds.

**Pull request merged**